pkg/controller/handler.go

```diff
@@ -180,40 +184,66 @@ func (p *Propeller) Handle(ctx context.Context, namespace, name string) error {
 			return nil
 		}
 	}
+	streak := 0
+	defer p.metrics.StreakLength.Add(ctx, float64(streak))
 
-	mutatedWf, err := p.TryMutateWorkflow(ctx, w)
-	if err != nil {
-		// NOTE We are overriding the deepcopy here, as we are essentially ingnoring all mutations
-		// We only want to increase failed attempts and discard any other partial changes to the CRD.
-		mutatedWf = RecordSystemError(w, err)
-		p.metrics.SystemError.Inc(ctx)
-	} else if mutatedWf == nil {
-		return nil
-	} else {
-		if !w.GetExecutionStatus().IsTerminated() {
-			// No updates in the status we detected, we will skip writing to KubeAPI
-			if mutatedWf.Status.Equals(&w.Status) {
-				logger.Info(ctx, "WF hasn't been updated in this round.")
-				return nil
+	maxLength := p.cfg.MaxStreakLength
+	if maxLength <= 0 {
+		maxLength = 1
+	}
+
+	for streak = 0; streak < maxLength; streak++ {
+		t := p.metrics.RoundTime.Start(ctx)
+		mutatedWf, err := p.TryMutateWorkflow(ctx, w)
+		if err != nil {
+			// NOTE We are overriding the deepcopy here, as we are essentially ingnoring all mutations
+			// We only want to increase failed attempts and discard any other partial changes to the CRD.
+			mutatedWf = RecordSystemError(w, err)
+			p.metrics.SystemError.Inc(ctx)
+		} else if mutatedWf == nil {
+			return nil
```

Comment:
Is this a legit case? that looks wrong... I know it existed... maybe at least add a log line here?